### ISSUE TITLE
Parse response body even if the response status code is not 2xx

### ISSUE THREAD

**Issue by jiajiawang** [ishkawa/APIKit#15]
Currently if the response status code is not 2xx, it won't parse the response body at all.

However sometimes the response body may contain useful info even if the response status code is not 2xx (mostly 4xx), for example, validation errors of object creation.

Can we please at least having an option to choose whether to parse the response body even if the response status code is not 2xx?

**Comment by ishkawa** [comment]
It's nice to create more detailed error using response. I have an idea.

- Add `API.responseErrorFromObject()` to give us a chance to handle error response.

```swift
public class API {
    public class func responseErrorFromObject(object: AnyObject) -> NSError
}
```

- Then, override this method in specific API class and return more detailed error.

```swift
public class YourAppAPI: API {
    public override class func responseErrorFromObject(object: AnyObject) -> NSError {
        if let message = (object as? NSDictionary)?["error"] as? String {
            let userInfo = [NSLocalizedDescriptionKey: message]
            return NSError(domain: "YourAppAPIErrorDomain", code: 40000, userInfo: userInfo) 
        } else {
            let userInfo = [NSLocalizedDescriptionKey: "unresolved error occurred."]
            return NSError(domain: "YourAppAPIErrorDomain", code: 40001, userInfo: userInfo) 
        }
    }
}
```

```json
{
    "error": "Some error occurred."
}
```

**Comment by ishkawa** [comment]
I will work on this, and APIKit 0.4 should contain this.

**Comment by ishkawa** [comment]
I have merged commits for this into develop/0.4, which will be released in a few days!

**Comment by ishkawa** [comment]
This is resolved in 0.4.0.
